app/views/provider_interface/provider_agreements/data_sharing_agreement.html.erb

```diff
@@ -19,97 +19,97 @@
     <h2 class="govuk-heading-m">Contents</h2>
     <ol class="app-contents-list__list">
       <li class="app-contents-list__list-item app-contents-list__list-item--parent">
-        <a class="govuk-link app-contents-list__link" href="#section-1">Section 1: Introduction</a>
+        <a class="app-contents-list__link" href="#section-1">Section 1: Introduction</a>
```

Comment:
Are these left as regular markup rather than converted to `govuk_link` because they are navigating to the same page?

Reply:
These could use the `govuk_link_to` helper, but would be far more verbose, i.e. instead of:

```html
<a class="govuk-link app-contents-list__link" href="#section-1">Section 1: Introduction</a>
```

we could write:

```erb
<%= govuk_link_to 'Section 1: Introduction', '#section-1', class: 'app-contents-list__link' %>
```

Hmmm, on reflection, maybe that’s fine? (Arguably `app-contents-list` would be a separate component, but beyond the scope of this PR)

Reply:
Updated to use `govuk_link_to` helper, as used elsewhere for contents lists.